doc/kernel/threads/lifecycle.rst

```diff
@@ -191,6 +207,41 @@ The following code has the same effect as the code segment above.
                     my_entry_point, NULL, NULL, NULL,
                     MY_PRIORITY, 0, K_NO_WAIT);
 
+User Mode Constraints
+---------------------
+
+This section only applies if :option:`CONFIG_USERSPACE` is enabled, and a user
+thread tries to create a new thread. The :c:func:`k_thread_create()` API is
+still used, but there are additional constraints which must be met or the
+calling thread will be terminated:
+
+* The calling thread must have permissions granted on both the child thread
+  and stack objects.
+
+* The child thread and stack objects must be in an uninitailized state,
+  i.e. it is not currently running and the stack memory is unused.
```

Comment:
I think this is probably cleared up later, but how is it enforced that the stack memory is "unused", and when does memory become "used"?

Reply:
stacks are tracked like other kernel objects, so initialization state is maintained in the associated struct _k_object